src/processors.zig

```diff
@@ -179,45 +182,50 @@ pub const CrossPageLinkProcessor = struct {
             }
         }
     }
-    fn parseAltText(
-        pctx: anytype,
-        alt_text: []const u8,
-        maybe_scale: ?[]const u8,
-    ) !void {
+    fn parseAltText(pctx: anytype, alt_text: []const u8, maybe_scale: ?[]const u8) !void {
         if (maybe_scale) |scale| {
-            if (parseResolution(scale)) |res| {
-                try pctx.out.print("alt=\"{s}\"", .{alt_text});
-                try pctx.out.print("width=\"{s}\"", .{res[0].?});
-                if (res[1]) |height| try pctx.out.print("height=\"{s}\"", .{height});
-                return;
+            const res = parseResolution(scale);
+            switch (res) {
```

Comment:
nitpick: this can be written like this, where the payload of the union is directly unpacked into the case block

```zig
switch (res) {
    .resolution => |resolution| {
        // use resolution instead of res.resolution
    },
    .alt_text => |given_alt_text| {
        // use given_alt_text
    },
```

Reply:
Oh, I wanted to do this but couldn't figure out where it wanted it. Thanks for catching that.